# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: fe6e63a2f914
# Full Hash: fe6e63a2f9141d9eeec41bfde52cc34b33c170fe
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-10-09 14:54 +0000
# Description:
#   Bug 1992536 - Use cookie creation time instead or PR_Now() to validate the expiration time, r=valentin,cookie-reviewers
# ==============================================================================

diff -r 0147060f88c2 -r fe6e63a2f914 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Thu Oct 09 14:36:53 2025 +0000
+++ b/netwerk/cookie/CookieService.cpp	Thu Oct 09 14:54:32 2025 +0000
@@ -588,7 +588,7 @@
                      dateHeader, true, isForeignAndNotAddon, mustBePartitioned,
                      storagePrincipalOriginAttributes.IsPrivateBrowsing(),
                      loadInfo->GetIsOn3PCBExceptionList(),
-                     CookieCommons::GetCurrentTimeFromChannel(aChannel));
+                     CookieCommons::GetCurrentTimeInUSecFromChannel(aChannel));
 
   if (!cookieParser.ContainsCookie()) {
     return NS_OK;